operator/bucketcontroller/webhook.go

```diff
@@ -0,0 +1,54 @@
+package bucketcontroller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// BucketValidator validates admission requests.
+type BucketValidator struct {
+	log logr.Logger
+}
+
+// ValidateCreate implements admission.CustomValidator.
+func (v *BucketValidator) ValidateCreate(_ context.Context, obj runtime.Object) error {
+	bucket := obj.(*exoscalev1.Bucket)
+	v.log.V(1).Info("Validate create", "name", bucket.Name)
+
+	providerConfigRef := bucket.Spec.ProviderConfigReference
+	if providerConfigRef == nil || providerConfigRef.Name == "" {
+		return fmt.Errorf("a bucket named %q requires a provider config reference",
+			bucket.Name)
```

Comment:
The returned error message is slightly ambiguous. Is the name referring to the bucket in the spec or the metadata name? Since the below will be wrapped in a kubernetes-generated error that already has the metadata.name, we don't need that info again.
```suggestion
		return fmt.Errorf(".spec.providerConfigRef.name is required")
```